src/object_store/src/object/s3.rs

```diff
@@ -392,6 +392,24 @@ impl ObjectStore for S3ObjectStore {
         })
     }
 
+    /// Returns a stream reading the object specified in `path`. If given, the stream starts at the
+    /// byte with index `start_pos` (0-based). As far as possible, the stream only loads the amount
+    /// of data into memory that is read from the stream.
+    async fn streaming_read(
+        &self,
+        path: &str,
+        start_pos: Option<usize>,
+    ) -> ObjectResult<Box<dyn AsyncRead + Unpin + Send + Sync>> {
+        fail_point!("s3_streaming_read_err", |_| Err(ObjectError::internal(
+            "s3 streaming read error"
+        )));
+
+        let req = self.obj_store_request(path, start_pos, None);
```

Comment:
Would it read the whole object since after `start_pos`?

Reply:
The HTTP library that is used internally has a buffer which is limited to 400 kB by default. Once the buffer is full, it will not download more data.